### ISSUE TITLE
Breadcrumbs show the UUID rather than asset name

### ISSUE THREAD

**Issue by lache-melvin** [msupply-foundation/open-msupply#4226]
## What went wrong? 😲

Slowed down for your viewing pleasure... it does get updated, and then overwritten!

https://github.com/msupply-foundation/open-msupply/assets/55115239/707ecb0a-8adc-44e4-a7ae-0597228fc6df

## Expected behaviour 🤔

Asset/Immunizations name stays in the breadcrumb!

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to Coldchain > Equipment
2. Click on an asset to go to the detail view
3. See breadcrumb: UUID!

## Your environment 🌱

- Open mSupply Version: 2.1 RC1
- Legacy mSupply Central Server Version:

- Platform: browser (chrome)

- Database type:

**Comment by lache-melvin** [comment]
WEIRDLY it works when you navigate from list view to detail page. But when you refresh the page, it goes back to being a uuid 👀

**Comment by mark-prins** [comment]
hm.. a re-render maybe, and the suffix call isn't made on the second one? is it worth refactoring the breadcrumb and see if that fixes this?

**Comment by CarlosNZ** [comment]
Existing related issues:
- https://github.com/msupply-foundation/open-msupply/issues/2118
- https://github.com/msupply-foundation/open-msupply/issues/4063

As Mark said, we should probably tackle that refactor and then see what remains of this problem.

**Comment by marthakeezy** [comment]
Just updated the issue to include another occurrence under Programs -> Immunizations on RC-2 :)

**Comment by andreievg** [comment]
Triage, putting this on hold, setting higher priority to this: https://github.com/msupply-foundation/open-msupply/issues/2118, once the refactor is done we can re-visiting this and re-check. Refactor to be prioritised for this RC

**Comment by marthakeezy** [comment]
Under Stock -> View Stock -> Log, can we also please show location's name instead of the ID?

![image](https://github.com/msupply-foundation/open-msupply/assets/153777484/c0f10a3c-3846-468c-ab24-951823b0d03f)

**Comment by marthakeezy** [comment]
Also just noticed under `Inventory -> View Stock`, the item's ID appears when navigating away from Details and then coming back to that tab. See below ⬇️ 

https://github.com/msupply-foundation/open-msupply/assets/153777484/baa4817f-059f-462d-81c4-a43c07d283ff

**Comment by lache-melvin** [comment]
> Under `Stock -> View Stock -> Log`, can we also please show location's name instead of the ID?
> 
> <img alt="Screenshot 2024-07-01 at 12 02 29 PM" width="1235" src="https://private-user-images.githubusercontent.com/153777484/344502466-8df2c3ac-27fd-478e-9608-077f3619e50b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk3OTM1OTYsIm5iZiI6MTcxOTc5MzI5NiwicGF0aCI6Ii8xNTM3Nzc0ODQvMzQ0NTAyNDY2LThkZjJjM2FjLTI3ZmQtNDc4ZS05NjA4LTA3N2YzNjE5ZTUwYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMVQwMDIxMzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NzU3Njg0OWJiNmNhOGNlYWRmMzQ2OWJmNWRlZmE3ZjIzYmFiYTQwZDlmNGI2YWUyYjgwOTA4OTY1ZTc4YmE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._AP50Ges33CvOXyzB0xCfOUGxFOYzTV4sIreF_v2pGc">

Thanks for eyes! This one would be a separate issue, probably raise as an enhancement? 🙏

**Comment by marthakeezy** [comment]
> > Under `Stock -> View Stock -> Log`, can we also please show location's name instead of the ID?
> > <img alt="Screenshot 2024-07-01 at 12 02 29 PM" width="1235" src="https://private-user-images.githubusercontent.com/153777484/344502466-8df2c3ac-27fd-478e-9608-077f3619e50b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk3OTM1OTYsIm5iZiI6MTcxOTc5MzI5NiwicGF0aCI6Ii8xNTM3Nzc0ODQvMzQ0NTAyNDY2LThkZjJjM2FjLTI3ZmQtNDc4ZS05NjA4LTA3N2YzNjE5ZTUwYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMVQwMDIxMzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NzU3Njg0OWJiNmNhOGNlYWRmMzQ2OWJmNWRlZmE3ZjIzYmFiYTQwZDlmNGI2YWUyYjgwOTA4OTY1ZTc4YmE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._AP50Ges33CvOXyzB0xCfOUGxFOYzTV4sIreF_v2pGc">
> 
> Thanks for eyes! This one would be a separate issue, probably raise as an enhancement? 🙏

#4325 :)

**Comment by lache-melvin** [comment]
Closed by #4326

**Comment by marthakeezy** [comment]
Tested in V2.1.0-RC5, no issues found :) Build test passed! ✅
